pkg/remote/kyma_synchronization_context.go

```diff
@@ -128,11 +129,33 @@ func (c *KymaSynchronizationContext) ensureRemoteNamespaceExists(ctx context.Con
 	return nil
 }
 
-func (c *KymaSynchronizationContext) CreateOrUpdateCRD(ctx context.Context, plural string) error {
+func UpdateKymaAnnotations(
```

Comment:
this function I suggest you design it more generic, you can create a string type attribute `CrdType` (either `KCP` or `SKR`) to indicate which crd annotation to update with,
```
UpdateKymaAnnotations(kyma *v1beta2.Kyma, crd *v1extensions.CustomResourceDefinition, crdType CrdType) {
     if kyma.Annotations == nil {
		kyma.Annotations = make(map[string]string)
     }
     kyma.Annotations[getAnotation(crd, crdType)] = strconv.FormatInt(crd.Generation, 10)
}
```
for that `getAnotation` function, you determine which annotation based on `crd.Spec.Names.Kind ` and `crdType`